.travis.yml

```diff
@@ -1,9 +1,17 @@
-# Sample .travis.yml for R projects
+# R for travis: see documentation at https://docs.travis-ci.com/user/languages/r
 
-language: r
-bioc_packages: IRanges
+language: R
+cache: packages
 warnings_are_errors: true
-sudo: required
+bioc_packages: IRanges
 
-after_success:
-  - Rscript -e 'covr::codecov()'
+matrix:
+  include:
+  - r: devel
+  - r: release
+    after_success:
+    - Rscript -e 'covr::codecov()'
+  - r: release
+    name: tidyr-devel
+    before_script: Rscript -e "remotes::install_github('tidyverse/tidyr')"
+  - r: oldrel
```

Comment:
This seems like a good basic build matrix (R devel, release, and oldrel). Technically, you should probably change the R version in DESCRIPTION to reflect the actively-tested versions. This is our policy in the Tidyverse (but we also test on a larger matrix, which is probably overkill for many other packages).

I've added a 2nd release with the released version of R that also checks against the dev version of tidyr. You might not want this in place all the time, but it's handy when a major dependency is nearing a release with breaking changes. I decided to demo this here in fuzzyjoin.